airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -233,10 +235,7 @@ private void replicate(final Path jobRoot,
 
       LOGGER.info("Waiting for source and destination threads to complete.");
       // CompletableFuture#allOf waits until all futures finish before returning, even if one throws an
-      // exception. So in order to handle exceptions from a future immediately without needing to wait for
-      // the other future to finish, we first call CompletableFuture#anyOf.
-      CompletableFuture.anyOf(readSrcAndWriteDstThread, readFromDstThread).get();
-      LOGGER.info("One of source or destination thread complete. Waiting on the other.");
+      // exception.
       CompletableFuture.allOf(readSrcAndWriteDstThread, readFromDstThread).get();
```

Comment:
I think the only future we really care about here is readFromDstThread, it might make sense to just wait for this one instead.